python-service/src/pipeline/event_processor.py

```
@@ -0,0 +1,146 @@
+    Messages are serialized by the backup scheduler using pickle so that
+    arbitrary BackupEvent subclasses (including those with custom __reduce__
+    methods added by platform plugins) round-trip correctly.
+
+    VULN-1 (Insecure deserialization — arbitrary code execution via pickle):
+    pickle.loads() executes arbitrary Python when the serialized object
+    contains a ``__reduce__`` or ``__reduce_ex__`` method.  Any process that
+    can write to the queue (a compromised microservice, a message-queue ACL
+    misconfiguration, or a crafted backup payload) can achieve RCE on the
+    consumer.  JSON + an explicit schema validator should be used instead.
+    """
+    return pickle.loads(raw_bytes)
```

Ruff (0.15.17)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)